### PR TITLE
[SPARK-40663][SQL] Migrate execution errors onto error classes: _LEGACY_ERROR_TEMP_2026-2050

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to migrate 25 execution errors onto temporary error classes with the prefix `_LEGACY_ERROR_TEMP_2026` to `_LEGACY_ERROR_TEMP_2050`.

The error classes are prefixed with `_LEGACY_ERROR_TEMP_` indicates the dev-facing error messages, and won't be exposed to end users.

### Why are the changes needed?

To speed-up the error class migration.

The migration on temporary error classes allow us to analyze the errors, so we can detect the most popular error classes.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

```
$ build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite"
$ build/sbt "test:testOnly *SQLQuerySuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@itholic Please, resolve conflicts.

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -3008,11 +3008,6 @@
       "<message>. If necessary set <ansiConfig> to false to bypass this error."
     ]
   },
-  "_LEGACY_ERROR_TEMP_2001" : {
-    "message" : [
-      "<message> If necessary set <ansiConfig> to false to bypass this error"
-    ]
-  },
```
Comment (itholic) [inline core/src/main/resources/error/error-classes.json:3015]: Addressed the https://github.com/apache/spark/pull/38104#discussion_r987730351.

@MaxGekk FYI

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 2]
```diff
@@ -299,7 +299,7 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
 
   def ansiIllegalArgumentError(message: String): SparkIllegalArgumentException = {
     new SparkIllegalArgumentException(
-      errorClass = "_LEGACY_ERROR_TEMP_2001",
+      errorClass = "_LEGACY_ERROR_TEMP_2000",
```
Comment (itholic) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:302]: and here

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @itholic.

**Pull request closed without merging**